jetty-server/src/main/java/org/eclipse/jetty/server/Dispatcher.java

```diff
@@ -250,28 +251,55 @@ public String toString()
         return String.format("Dispatcher@0x%x{%s,%s}",hashCode(),_named,_uri);
     }
 
-    private void commitResponse(ServletResponse response, Request baseRequest) throws IOException
+    @SuppressWarnings("Duplicates")
+    private void commitResponse(ServletResponse response, Request baseRequest) throws IOException, ServletException
     {
         if (baseRequest.getResponse().isWriting())
         {
             try
             {
+                // Try closing Writer first (based on knowledge in Response obj)
                 response.getWriter().close();
             }
-            catch (IllegalStateException e)
+            catch (IllegalStateException e1)
             {
-                response.getOutputStream().close();
+                try
+                {
+                    // Try closing OutputStream as alternate route
+                    // This path is possible due to badly behaving Response wrappers
+                    response.getOutputStream().close();
+                }
+                catch(IllegalStateException e2)
+                {
+                    MultiException me = new MultiException();
```

Comment:
Perhaps we can just add e2 as a suppressed exception to e1 rather than create a multi-exception?  We could just throw the ISE then.